Add future about calling init/postinit post-initialization

cassella opened the pull request:
Belatedly add future for #8991

lydia-duncan commented on test/types/records/init/init-post-postinit.future:
```diff
@@ -0,0 +1,2 @@
+design: Calls to .init and .postinit are allowed post-initialization
```
I would call this an error message future, since no one was arguing for allowing it.  So I would change this line to: `error message: Explicit calls to .init and .postinit should not be allowed`

cassella replied:
I'd started with something like that.  But when I checked TestSystem.rst, I saw it describes the "error message" category as for improvements to existing error messages.  (And I didn't read the discussion in the issue as "settled".  (Which is probably why I didn't submit a future for it at the time.))

I can make the change if you'd still like.

bradcray replied:
Paul's right that the "error message" tag is supposed to be about improving the wording/accuracy of an existing error message rather than creating an error message where none exists (which I might call a bug or a feature request depending on how we feel about it).

lydia-duncan replied:
I think I lean towards "bug" in this case, but I don't feel strongly about it.  I prefer either "bug" or "feature request" to "design"

cassella commented:
I changed the future category, adopted your description of it, and removed some trailing whitespace from the test itself that github hadn't shown.  start_test calls it a clean match against the .bad file.

I force-pushed since this didn't seem like history worth preserving.

lydia-duncan commented:
Also came back clean for me.  Thanks!
